### ISSUE TITLE
Exclude // TODOs from comment alignment

### ISSUE THREAD

**Issue by BalazsArva** [codecadwallader/codemaid#350]
### Environment
- Visual Studio version: 2013/2015 any edition
- CodeMaid version: 10.1
- Code language: C#
### Description

The cleanup command wraps all comments so that they don't exceed the specified upper limit of line width. I like this feature, however, I'd like it even more to be able to exclude certain comments, for example "// TODO" lines. This would be helpful because Visual Studio's Task List window only shows the actual line that starts with // TODO.

So if I have something like this:

```
// TODO: Remove
// this later.
```

Then the Task list window will only show "TODO: Remove", whereas I'd like to see the whole text.
### Steps to recreate
1. Add a long // TODO comment
2. Perform a cleanup
3. The TODO is now wrapped into several lines
4. View Task List window, only the first line of the comment is shown.
### Current behavior

Long TODO's are wrapped into several lines.
### Expected behavior

We should be able to exclude lines/comments from autowrapping based on patterns.

**Comment by codecadwallader** [comment]
Thanks for the suggestion to make an exception case for TODO comments wrapping when comment formatting is invoked upon them.  You mentioned being able to setup patterns for exclusions, were you thinking regular expression based patterns?

**Comment by BalazsArva** [comment]
Yes, though I am not sure if that wouldn't be too expensive. Maybe StartsWith() would do as well.

**Comment by DanielMKelly** [comment]
Possibly the Codemaid/Helpers/CodeCommentHelper.cs function GetCommentRegex should exclude TODO from registering as a comment and therefore be exempt from all formatting?

**Comment by willemduncan** [comment]
Agreed, same as with the `// Resharper Disable/Enable` comments. I would hard code a list and not expose it to configuration just yet.

**Comment by willemduncan** [comment]
> By default, Visual Studio includes the following tokens: `HACK`, `TODO`, `UNDONE`, `NOTE`. These are not case sensitive.

Let's add all of them while we're at it. There is no requirement for a semicolon `:`, either in documentation or in my tests on VS2017.

Which brings the problem, how to distinguish between this which should be left alone:
```cs
// Some piece of comment.
// TODO Do something very important with a long description that should not be wrapped.
// Maybe another comment.
```
And this which should be formatted as usual
```cs
// Writing some text using the
// todo word in the middle of it
// without it being something special.
```

So, how to distinguish between a "real" `TODO` and just the word used in a sentence? Same case for `HACK`, `UNDONE`, and `NOTE`. Actually even a bigger issue since those are more likely to appear in regular sentences.

**Comment by willemduncan** [comment]
This should be fixed with the merge of #545.
